atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,787 @@
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD.
+# See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import os
+import numpy as np
+import logging
+import h5py
+
+# from nomad.units import ureg
+from nomad.parsing.file_parser import FileParser
+from nomad.datamodel.metainfo.simulation.run import Run, Program  # TimeRun
+from nomad.datamodel.metainfo.simulation.method import (
+    Method, ForceField, Model, AtomParameters, ForceCalculations, NeighborSearching
+)
+from nomad.datamodel.metainfo.simulation.system import (
+    AtomsGroup
+)
+from nomad.datamodel.metainfo.simulation.calculation import (
+    Calculation, Energy, EnergyEntry, BaseCalculation
+)
+from nomad.datamodel.metainfo.workflow import (
+    Workflow
+)
+from nomad.datamodel.metainfo.simulation.workflow import (
+    EnsembleProperty, EnsemblePropertyValues, CorrelationFunction, CorrelationFunctionValues
+)
+from atomisticparsers.utils import MDParser
+from .metainfo.h5md import ParamEntry, CalcEntry, Author
+from nomad.units import ureg
+
+MOL = 6.022140857e+23
+
+
+class HDF5Parser(FileParser):
+    def __init__(self):
+        super().__init__(None)
+
+    @property
+    def filehdf5(self):
+        if self._file_handler is None:
+            try:
+                self._file_handler = h5py.File(self.mainfile, 'r')
+            except Exception:
+                self.logger.error('Error reading hdf5 file.')
+
+        return self._file_handler
+
+    def apply_unit(self, quantity, unit, unit_factor):
+
+        if quantity is None:
+            return
+        if unit:
+            unit = ureg(unit)
+            unit *= unit_factor
+            quantity *= unit
+
+        return quantity
+
+    def decode_hdf5_bytes(self, dataset, default):
+        if dataset is None:
+            return
+        elif type(dataset).__name__ == 'ndarray':
```

Comment:
can we use isinstance(dataset.?, np.ndarray)?

Reply:
indeed, that's better. Also fixed for the bool instance below. done.